test/unit/inbound/process-submitted-settlement.test.js

```diff
@@ -0,0 +1,137 @@
+const mockCommit = jest.fn()
+const mockRollback = jest.fn()
+const mockTransaction = {
+  commit: mockCommit,
+  rollback: mockRollback
+}
+
+jest.mock('../../../app/data', () => {
+  return {
+    sequelize:
+       {
+         transaction: jest.fn().mockImplementation(() => {
+           return { ...mockTransaction }
+         })
+       }
+  }
+})
+
+describe('process submitted settlement request', () => {
+  jest.mock('../../../app/inbound/get-existing-settlement')
+  const getExistingSettlement = require('../../../app/inbound/get-existing-settlement')
+
+  jest.mock('../../../app/inbound/save-settlement')
+  const saveSettlement = require('../../../app/inbound/save-settlement')
+
+  const processSubmittedSettlement = require('../../../app/inbound/process-submitted-settlement')
+
+  let settlement
+
+  beforeEach(() => {
+    settlement = {
+      sourceSystem: 'SITIAgri',
+      invoiceNumber: 'S123456789A123456V003',
+      frn: 1234567890,
+      currency: 'GBP',
+      value: 30000,
+      settlementDate: 'Fri Jan 21 2022 10:38:44 GMT+0000 (Greenwich Mean Time)',
+      reference: 'PY1234567',
+      settled: true
+    }
+
+    getExistingSettlement.mockResolvedValue(null)
+    saveSettlement.mockResolvedValue(null)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should call getExistingSettlement when a valid settlement is given ', async () => {
```

Comment:
my OCD is pointing out that all of yer test names have an empty space at the end